include/tvm/target/codegen.h

```diff
@@ -47,6 +47,37 @@ using runtime::TVMRetValue;
  */
 runtime::Module Build(IRModule mod, Target target);
 
+/*!
+ * \brief Serialize runtime module including its submodules
+ * \param mod The runtime module to serialize including its import tree.
+ * \param include_dso By default, include the info of DSOExportable modules. If disabled, an error
+ * will be raised when encountering DSO modules.
+ */
+std::string SerializeModuleToBytes(const runtime::Module& mod, bool include_dso = true);
```

Comment:
These functions should remain internal and avoid exposing for now

Reply:
`SerializeModuleToBytes` and `DeserailizeModuleFromBytes` are defined here because they are used in `src/node/structural_hash.cc`. Should I use these functions indirectly via packed function instead? 
```
TVM_REGISTER_REFLECTION_VTABLE(runtime::ModuleNode, ModuleNodeTrait)
    .set_creator([](const std::string& blob) {
      runtime::Module rtmod = codegen::DeserializeModuleFromBytes(blob);
      return RefToObjectPtr::Get(rtmod);
    })
    .set_repr_bytes([](const Object* n) -> std::string {
      const auto* rtmod = static_cast<const runtime::ModuleNode*>(n);
      return codegen::SerializeModuleToBytes(GetRef<runtime::Module>(rtmod), /*include_dso*/ false);
    });
```

Reply:
we should avoid serialization and implementing structural hash/equal in module for now, mainly because they are not useful for all instead rely on special checking

Reply:
Based on offline discussion, we agreed to keep `SerializeModuleToBytes` and `DeserailizeModuleFromBytes` as they are now.